### ISSUE TITLE
Discriminative fine-tuning for new (added) words

### ISSUE THREAD

**Issue by Aktsvigun** [huggingface/transformers#4823]
Good afternoon,
I have a question about the way of changing the learning rate for some parameters of the model. Let's say we have a BERT model and we have added a few new tokens. Consequently, we need to resize the embedding layer and initialize the embeddings for new words randomly. Meanwhile the learning rate of 3e-5 is used to train the model (otherwise we "overjump" the global minimum). If we use this LR for the embeddings of the new words as well, they will hardly change and thus will be close to random, thereby, the reasonable way is to change the learning rate only for their embeddings (as it is done, for instance, in ULMFiT). The question is: is there a simple way to do it in HuggingFace? Or probably are there some examples of doing it? Thanks in advance!

**Comment by Aktsvigun** [comment]
Finally found the solution. The optimizer should look like this:
```
optim.SGD([
    {'params': model.base.parameters()},
    {'params': model.classifier.parameters(), 'lr': 1e-3}
], lr=1e-2, momentum=0.9))
```

**Comment by Aktsvigun** [comment]
However, this approach does not work when I need to specify another LR not for the whole layer, but only for a few weights from it. Hence, the issue is still open. @LysandreJik @patrickvonplaten will be very grateful if you could help.

**Comment by patrickvonplaten** [comment]
Hi @Aktsvigun, 

Thanks for your detailed question! I'm not super familiar with these kind of specifics for training, but I'm not sure that this is even possible in PyTorch. 

Also, did you try to fine-tune the model normally as well without setting a specific learning rate to only one parameter of a layer? I would expect normal fine-tuning to also work quite well since the gradient (independently of the lr) for the newly added weight will be quite high and thus change significantly even for the same learning rate for all parameters.

**Comment by Aktsvigun** [comment]
@patrickvonplaten thank you for the answer! 

I did try with a simple LR but what urged me to the question is the difference in the results of a language model, fine-tuned without adding new words and the one with changed vocab. I took a small dataset (_~ 25 000 sentences_) and fine-tuned both models with `lr = 3e-5` (pretty standard as I know) and `num_epochs = 10`. The model without new words had an eval loss (on the validation sample) of **2.55**, while the loss for the one with 250 new words (not much really when the vocab size equals 50250) equaled 2.98. The words are not so popular among the dataset itself, however, such a great difference can be caused only by underfitting in my view.
